iib/workers/tasks/utils.py

```diff
@@ -68,6 +73,80 @@ def inner(*args, **kwargs):
     return wrapper
 
 
+def reset_docker_config():
+    """Create a symlink from ``iib_docker_config_template`` to ``~/.docker/config.json``."""
+    conf = get_worker_config()
+    docker_config_path = os.path.join(os.path.expanduser('~'), '.docker', 'config.json')
+
+    try:
+        log.debug('Removing the Docker config at %s', docker_config_path)
+        os.remove(docker_config_path)
+    except FileNotFoundError:
+        pass
+
+    if os.path.exists(conf.iib_docker_config_template):
+        log.debug(
+            'Creating a symlink from %s to %s', conf.iib_docker_config_template, docker_config_path
+        )
+        os.symlink(conf.iib_docker_config_template, docker_config_path)
+
+
+@contextmanager
+def set_registry_token(token, container_image):
+    """
+    Configure authentication to the registry that ``container_image`` is from.
+
+    This context manager will reset the authentication to the way it was after it exits. If
+    ``token`` is falsy, this context manager will do nothing.
+
+    :param str token: the token in the format of ``username:password``
+    :param str container_image: the pull specification of the container image to parse to determine
+        the registry this token is for.
+    :return: None
+    :rtype: None
+    """
+    if not token:
```

Comment:
What happens if request A uses an overwrite token and the clean up fails to execute. When a new request B comes along, which doesn't use an overwrite token, the old authfile will be used, no?

Reply:
`_cleanup` is always called at the start up of the Celery tasks, which resets the Docker config to the original.